packages/plugin-ext/src/main/browser/menus/menus-contribution-handler.ts

```diff
@@ -266,36 +266,50 @@ export class MenusContributionPointHandler {
-        this.registerMenuAction(menuPath, menu, {
-            execute: () => this.commands.executeCommand(menu.command, selectedResource()),
-            isEnabled: () => this.commands.isEnabled(menu.command, selectedResource()),
-            isVisible: () => this.commands.isVisible(menu.command, selectedResource())
-        });
+        this.registerMenuAction(menuPath, menu, command => ({
+            execute: () => this.commands.executeCommand(command, selectedResource()),
+            isEnabled: () => this.commands.isEnabled(command, selectedResource()),
+            isVisible: () => this.commands.isVisible(command, selectedResource())
+        }));
     }
 
-    protected registerMenuAction(menuPath: MenuPath, menu: Menu, handler: CommandHandler): void {
-        const commandId = this.createSyntheticCommandId(menu, { prefix: '__plugin.menu.action.' });
+    protected registerMenuAction(menuPath: MenuPath, menu: Menu, handler: (command: string) => CommandHandler): void {
+        const commandId = this.createSyntheticCommandId(menu.command, { prefix: '__plugin.menu.action.' });
         const command: Command = { id: commandId };
-        this.commands.registerCommand(command, handler);
+        this.commands.registerCommand(command, handler(menu.command));
+
+        const altId = menu.alt && this.createSyntheticCommandId(menu.alt, { prefix: '__plugin.menu.action.' });
+        const alt: Command | undefined = altId && { id: altId } || undefined;
+        if (alt) {
+            this.commands.registerCommand(alt, handler(menu.alt!));
+        }
 
         const { when } = menu;
         const [group = '', order = undefined] = (menu.group || '').split('@');
-        const action: MenuAction = { commandId, order, when };
+        const action: MenuAction = { commandId, alt: altId, order, when };
         const inline = /^inline/.test(group);
         menuPath = inline ? menuPath : [...menuPath, group];
         this.menuRegistry.registerMenuAction(menuPath, action);
 
         this.onDidRegisterCommand(menu.command, pluginCommand => {
             command.category = pluginCommand.category;
-            action.label = pluginCommand.label;
+            command.label = pluginCommand.label;
             if (inline) {
-                action.icon = pluginCommand.iconClass;
+                command.iconClass = pluginCommand.iconClass;
             }
         });
+        if (alt && menu.alt) {
+            this.onDidRegisterCommand(menu.alt, pluginCommand => {
+                alt.category = pluginCommand.category;
+                alt.label = pluginCommand.label;
+                if (inline) {
+                    alt.iconClass = pluginCommand.iconClass;
+                }
+            });
+        }
     }
 
-    protected createSyntheticCommandId(menu: Menu, { prefix }: { prefix: string }): string {
-        const command = menu.command;
+    protected createSyntheticCommandId(command: string, { prefix }: { prefix: string }): string {
```

Comment:
Please add this breaking change to the changelog. Thanks!

Reply:
ok